app/src/main/java/com/example/itbook/base/BaseFragment.kt

```diff
@@ -49,6 +52,21 @@ abstract class BaseFragment : Fragment() {
             ?.commit()
     }
 
+    private fun initDialog() {
+        loadingDialogFragment = LoadingDialogFragment()
+        loadingDialogFragment?.isCancelable = false
+    }
+
+    fun showLoading(isShow: Boolean) {
+        loadingDialogFragment?.run {
+            if (isShow) {
+                activity?.let { show(it.supportFragmentManager, "") }
+            } else {
+                dismiss()
+            }
+        }
+    }
```

Comment:
em nên tách ra làm 2 hàm show và hide để mỗi hàm thực hiện 1 nhiệm vụ 
Gắng hạn chế phải check null nhiều lần, và để các scope function lồng nhau 
```suggestion
    fun showLoading(isShow: Boolean) {
            if (isShow) {
                activity?.supportFragmentManager?.let{  loadingDialogFragment.show(it.TAG) }
            } else {
                it.dismiss()
            }
        }
    }
```